suthread.cpp

```diff
@@ -109,15 +109,19 @@ Value ThreadClass::call(Value self, Value member,
 		{
 		NOARGS("Thread.List()");
 		SuObject* list = new SuObject();
-		//TODO implement Thread.List()
+		Fibers::foreach_fiber_info(
+			[list](gcstring name, const char* status) { list->putdata(name.str(), status); });
 		return list;
 		}
 	else if (member == Name)
 		{
 		if (nargs > 1)
 			except("usage: Thread.Name(name = false)");
-		//TODO implement Thread.Name()
-		return "";
+		if (nargs == 0)
+			return Fibers::get_name().str();
+		const char* name = ARG(0).str();
+		gcstring fiber_name = Fibers::set_name(name);
+		return fiber_name.str();

```

Comment:
I missed that

I would use "new SuString(name)"

But not status since there are only two statuses and creating symbols for them avoids allocating a new string each time. (although very minor)